File: Python Assignments/readability.py
# TODO

import cs50

def main():

    text = cs50.get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    grade = round(0.0588 * 100*letters/words - 0.296 * 100*sentences/words - 15.8)

    print(grade)
    print(letters)
    print(words)
    print(sentences)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print("Grade " + str(grade))

def count_letters(txt):

    letter_count = 0
    for i in range (len(txt)):
        if txt[i].isalpha():
            letter_count += 1

    return letter_count

def count_words(txt):

    word_count = 1

    for i in range(len(txt)):
        if txt[i].isspace():
            word_count+=1

    return word_count

def count_sentences(txt):

    sentence_count = 0

    for i in range(len(txt)):
        if txt[i] == "." or txt[i] == "?" or txt[i] == "!":
            sentence_count += 1
    return sentence_count

main()

File: C assignments/readability.c
//determines the reading level of the text provided by user. usage - ./readability

#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int words = count_words(text);
    int letters = count_letters(text);
    int sentences = count_sentences(text);
    float grade_actual = 5.88*letters/words - 29.6*sentences/words - 15.8;
    int grade = round(grade_actual);

    printf("%i\n", letters);
    printf("%i\n", words);
    
    if (grade_actual < 1)
    {
        printf("Before Grade 1\n");
    } else if (grade_actual > 16)
    {
        printf("Grade 16+\n");
    } else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    int count = 0;
    int text_length = strlen(text);
    for (int i = 0; i < text_length; i++)
    {
        if (isalpha(text[i]) || isdigit(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1;
    int text_length = strlen(text);
    for (int i = 0; i < text_length; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    int text_length = strlen(text);
    for (int i = 0; i < text_length; i++)
    {
        if (text [i] == '.' || text [i] == '?' || text [i] == '!')
        {
            count++;
        }
    }
    return count;
}
